migration/src/main/java/kin/sdk/migration/MigrationManager.java

```diff
@@ -1,60 +1,404 @@
-                            IKinVersionProvider kinVersionProvider, IWhitelistService whitelistService) {
-        this(context, appId, networkUrl, networkId, issuer, kinVersionProvider, whitelistService, null);
-    }
-
-    public MigrationManager(Context context, String appId, String networkUrl, String networkId, String issuer,
-                            IKinVersionProvider kinVersionProvider, IWhitelistService whitelistService, String storeKey) {
-        this.context = context;
-        this.appId = appId;
-        this.networkUrl = networkUrl;
-        this.networkId = networkId;
-        this.issuer = issuer;
-        this.kinVersionProvider = kinVersionProvider;
-        this.whitelistService = whitelistService;
-        this.storeKey = storeKey;
-    }
-
-    public IKinClient initMigration() {
-        IKinClient kinClient;
-        boolean kinSdkVersion = kinVersionProvider.isKinSdkVersion();
-        storeKey = (storeKey != null) ? storeKey : "";
-        if (kinSdkVersion) {
-            Environment environment = new Environment(networkUrl, networkId);
-            kinClient = new KinClientSdkImpl(context, environment, appId, whitelistService, storeKey);
-        } else {
-            ServiceProvider environment = new ServiceProvider(networkUrl, networkId) {
-                @Override
-                protected String getIssuerAccountId() {
-                    return issuer;
-                }
-            };
-            kinClient = new KinClientCoreImpl(context, environment, storeKey);
-        }
-        return kinClient;
-    }
+	private static final String TAG = MigrationManager.class.getSimpleName();
+	private final static String MIGRATION_MODULE_PREFERENCE_FILE_KEY = "MIGRATION_MODULE_PREFERENCE_FILE_KEY";
+	private final static String MIGRATION_COMPLETED_KEY = "MIGRATION_COMPLETED_KEY";
+	private final static int TIMEOUT = 30;
+	private final static int MAX_RETRIES = 3;
+	private static final String URL_MIGRATE_ACCOUNT_SERVICE = "https://migration-devplatform-playground.developers.kinecosystem.com/migrate?address=";
+
+	private Context context;
+	private final String appId;
+	private MigrationNetworkInfo migrationNetworkInfo;
+	private IKinVersionProvider kinVersionProvider;
+	private String storeKey;
+	private Handler handler;
+	private OkHttpClient okHttpClient;
+	private volatile AtomicBoolean inMigrationProcess; // defence against multiple calls
+
+	public MigrationManager(@NonNull Context context, @NonNull String appId,
+		@NonNull MigrationNetworkInfo migrationNetworkInfo,
+		@NonNull IKinVersionProvider kinVersionProvider) {
+		this(context, appId, migrationNetworkInfo, kinVersionProvider, "");
+	}
+
+	public MigrationManager(@NonNull Context context, @NonNull String appId,
+		@NonNull MigrationNetworkInfo migrationNetworkInfo,
+		@NonNull IKinVersionProvider kinVersionProvider, @NonNull String storeKey) {
+		this.context = context;
```

Comment:
I ahve no control over what context the developer will give us(although there is a code which can validate if a context is the application context).
I can change the name and the documentation so it will be clear for the developer.
What do you think?